go/ct/common/hash256_test.go

```diff
@@ -0,0 +1,98 @@
+//
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use
+// of this software will be governed by the GNU Lesser General Public Licence v3
+//
+
+package common
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/Fantom-foundation/Tosca/go/vm"
+)
+
+func TestHash256_EqualWhenContainingSameContent(t *testing.T) {
+	b1 := NewHash256(vm.Hash{1})
+	b2 := b1
+	b3 := NewHash256(vm.Hash{2})
+
+	if !b1.Equal(b2) {
+		t.Errorf("instances are not equal, got %v and %v", b1, b2)
+	}
+
+	if b1.Equal(b3) {
+		t.Errorf("instances are equal, got %v and %v", b1, b3)
+	}
+}
+
+func TestHash256_AssignmentProducesEqualValue(t *testing.T) {
+	b1 := NewHash256(vm.Hash{1})
+	b2 := b1
+
+	if !b1.Equal(b2) {
+		t.Errorf("assigned value is not equal: %v vs %v", b1, b2)
+	}
+}
+
+const (
+	hash0 = "[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]"
+	hash1 = "[1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]"
```

Comment:
Since they are only used in the function below their definition could be moved inside the function.